File: tests/fixtures/negative/android-lint/SecureRandom.java
package com.example;

import java.security.SecureRandom;

class JavaTokenGenerator {
    long byteArraySeed(byte[] seedBytes) {
        SecureRandom rng = new SecureRandom();
        rng.setSeed(seedBytes);
        return rng.nextLong();
    }

    static class Seeder {
        void setSeed(long seed) {}
    }

    void unrelatedSetSeed() {
        Seeder rng = new Seeder();
        rng.setSeed(1L);
    }
}

File: tests/fixtures/negative/android-lint/SecureRandom.kt
package com.example

import java.security.SecureRandom
import java.util.concurrent.ThreadLocalRandom

class TokenGenerator {
    fun generateToken(): Long {
        val rng = SecureRandom()
        return rng.nextLong()
    }

    fun nextInt(): Int {
        return ThreadLocalRandom.current().nextInt()
    }

    fun byteArraySeed(seedBytes: ByteArray): Long {
        val rng = SecureRandom()
        rng.setSeed(seedBytes)
        return rng.nextLong()
    }

    fun kotlinRandom(): Int {
        val rng = kotlin.random.Random(1234)
        return rng.nextInt()
    }

    fun comment() {
        // Random()
        val unrelated = "Random("
        println(unrelated)
    }
}
